### DISCUSSION TITLE
Open local file with PWA

### DISCUSSION THREAD

**Discussion by PippoBaudo-ai** [filips123/PWAsForFirefox#562] [Q&A]
Is it possible to open a local file with the installed PWA?
As an example, consider https://excalidraw.com. Once installed it would be nice to be able to open a local .excalidraw file like

```
firefoxpwa site launch <id> /path/to/file.excalidraw
```

This is possible with chrome/chromium,

```
chromium --profile-directory=... --app-id=... /path/to/file.excalidraw
```

but it's way more cumbersome because it does not have a proper CLI tool.
This way it is also possible to add the PWA as the default app to open a .excalidraw file.

**Comment by filips123** [comment]
This is not possible as excalidraw.com uses a non-standard `file_handlers`/`launchQueue` API that is only supported in Chrome. I might add support for this API in the future, but I'm not sure yet.

You can instead just open excalidraw PWA and load the file from the normal file dialog.
